# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1281516931 -7200
#      Wed Aug 11 10:55:31 2010 +0200
# Node ID cc2bc3f46cd4e1682d7da651d66a5d6fc722b234
# Parent  aef238768f25f931f82444494115651003d228ff
fix assignment bug with lazy indices

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-08-11  Jaroslav Hajek  <highegg@gmail.com>
+
+	* ov-lazy-idx.h (octave_lazy_index::subsasgn): Remove override. Leave
+	it to the defualt numeric conversion mechanism.
+
 2010-08-10  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (Fdiff): Allow logical type.
 
 2010-08-09  Rik <octave@nomad.inbox5.com>
 
 	* oct-parse.yy (parse_fcn_file): Restore parsing of help text
 	accidentally lost in previous changeset.
diff --git a/src/ov-lazy-idx.h b/src/ov-lazy-idx.h
--- a/src/ov-lazy-idx.h
+++ b/src/ov-lazy-idx.h
@@ -74,21 +74,16 @@ public:
   octave_value subsref (const std::string& type,
                         const std::list<octave_value_list>& idx)
     { return make_value ().subsref (type, idx); }
 
   octave_value_list subsref (const std::string& type,
                              const std::list<octave_value_list>& idx, int)
     { return subsref (type, idx); }
 
-  octave_value subsasgn (const std::string& type,
-                         const std::list<octave_value_list>& idx,
-                         const octave_value& rhs)
-    { return make_value ().subsasgn (type, idx, rhs); }
-
   octave_value do_index_op (const octave_value_list& idx,
                             bool resize_ok = false)
     { return make_value ().do_index_op (idx, resize_ok); }
 
   dim_vector dims (void) const { return index.orig_dimensions (); }
 
   octave_idx_type numel (void) const { return index.length (0); }
 
